File: daily/rust/Dec-16-valid-anagram.rs
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len() {
            return false;
        }
        let mut buffers = vec![0; 26];
        let mut buffert = vec![0; 26];
        s.chars()
            .for_each(|c| buffers[c as usize - 'a' as usize] += 1);
        t.chars()
            .for_each(|c| buffert[c as usize - 'a' as usize] += 1);
        if buffers == buffert {
            return true;
        }
        false
    }
}

File: daily/cpp/Dec-16-valid-anagram.cpp
#include <string>
using namespace std;

class Solution {
 public:
  bool isAnagram(string s, string t) {
    if (s.size() != t.size()) return false;
    vector<int> buffers(26, 0);
    vector<int> buffert(26, 0);

    for (char c : s) {
      buffers[c - 'a']++;
    }
    for (char c : t) {
      buffers[c - 'a']++;
    }
    if (buffers == buffert) return true;
    return false;
  }
};
